### PR TITLE
Capella changes

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR Description
Update Web3Signer to sign block for Capella spec.

- Update OpenApi spec
- Introduced manual fork overridden cli option
- Update Acceptance Test
- Update Teku library to 23.2.0

## Fixed Issue(s)

#682 
## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/Eth2BlockSigningAcceptanceTest.java [hunk 1]
```diff
@@ -88,14 +88,18 @@ void signAndVerifyLegacyBlockSignature() throws Exception {
...
-  @Test
-  void emptyBlockRequestReturnsBadRequestStatus() throws JsonProcessingException {
-    final Eth2BlockSigningRequestUtil util =
-        new Eth2BlockSigningRequestUtil(SpecMilestone.BELLATRIX);
-    setupEth2Signer(Eth2Network.MINIMAL, SpecMilestone.BELLATRIX);
+  @ParameterizedTest(
+      name = "#{index} - Empty block request for spec {0} should return bad request status")
+  @EnumSource(
+      value = SpecMilestone.class,
+      names = {"BELLATRIX", "CAPELLA"})
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/Eth2BlockSigningAcceptanceTest.java:95]: Why does this only apply to bellatrix and capella? should also include phase0 and altair?

File: core/src/main/resources/openapi-specs/eth2/signing/schemas.yaml [hunk 2]
```diff
@@ -464,6 +465,19 @@ components:
       required:
         - version
         - block_header
+    BlockRequestCapella:
+      type: object
+      properties:
+        version:
+          type: string
+          enum:
+            - CAPELLA
```
Comment (jframe) [inline core/src/main/resources/openapi-specs/eth2/signing/schemas.yaml:474]: Annoying that we need to define this just for the enum value. But think we have to do this way with the current API.

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/Eth2BlockSigningAcceptanceTest.java (hunk 1 above)
Comment (usmansaleem) [reply acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/Eth2BlockSigningAcceptanceTest.java]: good point, updated it to cover phase0 and altair as well.

**Review by jframe — APPROVED** [review]
LGTM

**Pull request merged**